static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-epic-liveblog.js

```diff
@@ -51,24 +61,27 @@ const setupViewTracking = (el, test) => {
     });
 };
 
-const addEpicToBlocks = (epicHtml, test) =>
-    fastdom.write(() => {
-        const $blocksToInsertEpicAfter = $(`.${INSERT_EPIC_AFTER_CLASS}`);
-
-        $blocksToInsertEpicAfter.each(el => {
-            const $epic = $.create(epicHtml);
-
-            $epic.insertAfter(el);
-            $(el).removeClass(INSERT_EPIC_AFTER_CLASS);
-
-            setEpicLiveblogEntryTimeData(
-                $epic[0],
-                getLiveblogEntryTimeData(el)
-            );
-
-            setupViewTracking(el, test);
+const addEpicToBlocks = (
+    epicHtml: string,
+    test: ContributionsABTest
+): Promise<void> =>
+    fastdom
+        .read(() => $(`.${INSERT_EPIC_AFTER_CLASS}`))
+        .then($blocksToInsertEpicAfter => {
+            $blocksToInsertEpicAfter.each(el => {
+                fastdom
+                    .read(() => getLiveblogEntryTimeData(el))
```

Comment:
I think rather than doing that, I'd modify `getLiveblogEntryTimeData` to return a fastdom Promise, so you can write:

```js
getLiveblogEntryTimeData(el)
  .then(timeData => { ... })
```

which reads a bit better. What do you think?